### ISSUE TITLE
Strange delay on SSH connection caused by seemingly innocuous change

### ISSUE THREAD

**Issue by Stono** [istio/istio#27941]
**Bug description**

For context, we have a `Pod` like this:
```
    ports:
    - containerPort: 9080
      name: http-web-admin
      protocol: TCP
    - containerPort: 8317
      name: http-web
      protocol: TCP
    - containerPort: 8318
      name: tcp-sftp
      protocol: TCP
```

And a `Sidecar` like this (note, only referencing the `http` ports):
```
apiVersion: networking.istio.io/v1beta1
kind: Sidecar
metadata:
  name: vds-acceptor
  namespace: vds-acceptor
spec:
  ingress:
  - defaultEndpoint: 127.0.0.1:8317
    port:
      name: http-app
      number: 8317
      protocol: HTTP
  - defaultEndpoint: 127.0.0.1:9080
    port:
      name: http-admin
      number: 9080
      protocol: HTTP
```

And the following annotation on the `PodSpec`:

```
annotations:
  traffic.sidecar.istio.io/includeInboundPorts: 8317,9080
```

And everything is fine.  

However, we thought that the annotation `traffic.sidecar.istio.io/includeInboundPorts: 8317,9080` is redundant due to the more recently added `Sidecar` ingress config shown above.

When we remove the annotation the service still works, but we start getting a delay (feels like around `500ms` but does have some variance):

```
❯ telnet
telnet> o
(to) 10.193.32.2 22
Trying 10.193.32.2...
Connected to 10.193.32.2.
Escape character is '^]'.

<-- delay here

SSH-2.0-SSHD-CORE-1.4.0
```

This causes us issues as we have services expecting to see `SSH-2.0-SSHD-CORE-1.4.0` without such a delay.  It feels like protocol sniffing, but we have `enableProtocolSniffingForInbound` and `enableProtocolSniffingForOutbound` set to `false` and also just for the sake of sanity we tried setting `protocolDetectionTimeout: 1ms`.

I'm also slightly confused, why with or without the annotation we get a listener?
```
❯ istioctl proxy-config listeners vds-acceptor-79878c5667-qw7sh.vds-acceptor
ADDRESS           PORT      TYPE
10.198.29.167     8317      HTTP
10.198.29.167     9080      HTTP
10.192.37.138     22        TCP
```

We cannot see any differences in the `proxy-config listeners|clusters` between a deployment with or without this annotation, so we're stumped.  Adding the annotation back removes the delay.

[ ] Docs
[ ] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[x] User Experience
[ ] Developer Infrastructure

**Expected behavior**
Removing the annotation to not to introduce an arbitrary tcp delay.

**Steps to reproduce the bug**
As above

**Version (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version` if you used Helm)**
1.5.10

**How was Istio installed?**
Helm

**Environment where bug was observed (cloud vendor, OS, etc)**
GKE

**Comment by howardjohn** [comment]
traffic.sidecar.istio.io/includeInboundPorts only impacts the iptables so it would make sense it doesn't impact the XDS config

**Comment by Stono** [comment]
@howardjohn we're unable to reproduce this on 1.6.  
The issue was actually raised against a 1.5 control plane.  On 1.5 we've worked around it with the `includeInboundPorts` annotation, and obviously don't expect you to be releasing a fix for 1.5 so will close it off.
